swarm/storage/resource.go

```diff
@@ -687,3 +769,33 @@ func getNextPeriod(start uint64, current uint64, frequency uint64) uint32 {
 	period := blockdiff / frequency
 	return uint32(period + 1)
 }
+
+func ToSafeName(name string) (string, error) {
+	validname, err := idna.ToASCII(name)
+	if err != nil {
+		return "", err
+	}
+	return validname, nil
+}
+
+// check that name identifiers contain valid bytes
+func isSafeName(name string) bool {
+	if name == "" {
+		return false
+	}
+	validname, err := idna.ToASCII(name)
+	if err != nil {
+		return false
+	}
+	return validname == name
+}
+
+// convenience for creating signature hashes of update data
+func (self *ResourceHandler) keyDataHash(key Key, data []byte) common.Hash {
+	self.hashLock.Lock()
+	defer self.hashLock.Unlock()
+	self.hasher.Reset()
```

Comment:
Consider using [`sync.Pool`](https://golang.org/pkg/sync/#Pool) to make this more concurrent.

Reply:
You mean have several hashers in the pool?